lib/core/src/Cardano/Wallet/Primitive/Types/UTxOIndex/Internal.hs

```diff
@@ -167,25 +166,26 @@ data UTxOIndex = UTxOIndex
 
 instance NFData UTxOIndex
 
+instance Semigroup UTxOIndex where
+    a <> b = fromSequence (toList a <> toList b)
+
+instance Monoid UTxOIndex where
+    mempty = UTxOIndex
+        { assetsAll = Map.empty
+        , assetsSingleton = Map.empty
+        , coins = Set.empty
+        , balance = mempty
+        , utxo = Map.empty
```

Comment:
In the above expression, the type of each of the field is obvious, except for `balance`.

I reckon this is much clearer:
```hs
    mempty = UTxOIndex
        { assetsAll = Map.empty
        , assetsSingleton = Map.empty
        , coins = Set.empty
        , balance = TokenBundle.empty
        , utxo = Map.empty
        }
 ```